x/leverage/keeper/invariants.go

```diff
@@ -0,0 +1,331 @@
+			denom := types.DenomFromKeyWithAddress(key, collateralPrefix)
+			// remove prefix | denom and null-terminator
+			address := types.AddressFromKey(key, collateralPrefix)
+
+			amount := sdk.ZeroInt()
+			if err := amount.Unmarshal(val); err != nil {
+				count++
+				msg += fmt.Sprintf("\t%s - %s received an error while Unmarshal the byte %+v\n", denom, address.String(), val)
+				return nil
+			}
+
+			if !amount.IsPositive() {
+				count++
+				msg += fmt.Sprintf("\t%s - %s collateral amount %s is not positive\n", denom, address.String(), amount.String())
+			}
+			return nil
+		})
+
+		if err != nil {
+			msg += fmt.Sprintf("\tSome error occurred while iterating through the collateral amount %+v\n", err)
+		}
+
+		broken := count != 0
+
+		return sdk.FormatInvariant(
+			types.ModuleName, routeCollateralAmount,
+			fmt.Sprintf("number of not positive collateral amount found %d\n%s", count, msg),
+		), broken
+	}
+}
+
+// BorrowAmountInvariant checks that borrow amounts have all positive values
+func BorrowAmountInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			msg   string
+			count int
+		)
+
+		loanKeyPrefix := types.CreateLoanKeyNoAddress()
+
+		// Iterate through all denoms which have an borrow amount stored
+		// in the keeper. If a token is registered but its borrow amount is
+		// not positive or it has some error doing the unmarshal it
+		// adds the denom and address invariant count and message description
+		err := k.iterate(ctx, loanKeyPrefix, func(key, val []byte) error {
+			// remove prefix | lengthPrefixed(addr) and null-terminator
+			denom := types.DenomFromKeyWithAddress(key, loanKeyPrefix)
+			// remove prefix | denom and null-terminator
+			address := types.AddressFromKey(key, loanKeyPrefix)
+
+			amount := sdk.ZeroInt()
+			if err := amount.Unmarshal(val); err != nil {
+				count++
+				msg += fmt.Sprintf("\t%s - %s received an error while Unmarshal the byte %+v\n", denom, address.String(), val)
+				return nil
+			}
+
+			if !amount.IsPositive() {
+				count++
+				msg += fmt.Sprintf("\t%s - %s borrow amount %s is not positive\n", denom, address.String(), amount.String())
+			}
+			return nil
+		})
+
+		if err != nil {
+			msg += fmt.Sprintf("\tSome error occurred while iterating through the borrow amount %+v\n", err)
+		}
+
+		broken := count != 0
+
+		return sdk.FormatInvariant(
+			types.ModuleName, routeBorrowAmount,
+			fmt.Sprintf("number of not positive borrow amount found %d\n%s", count, msg),
+		), broken
+	}
+}
+
+// BorrowAPYInvariant checks that Borrow APY have all positive values
+func BorrowAPYInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			msg   string
+			count int
+		)
+
+		borrowAPYprefix := types.CreateBorrowAPYKeyNoDenom()
+
+		// Iterate through all denoms which have an Borrow APY stored
+		// in the keeper. If a token is registered but its borrow APY is
```

Comment:
```suggestion
		// in the keeper. If a borrow APY is registered but it is
```
The reason for this wording change is, "if a token is registered" would probably refer to the Token Registry (at `KeyPrefixRegisteredToken`), but we are only iterating through registered APYs.